### DISCUSSION TITLE
Catch Audit Event Save Exceptions

### DISCUSSION THREAD

**Discussion by shoregl** [thepirat000/Audit.NET#626] [Q&A]
I need to catch exceptions (and then send a notification) when an audit event fails to be saved.  What are the good options for accomplishing this, preferably with as little code as possible.

**Comment by thepirat000** [comment]
I think the easiest way would be to create a custom data provider that inherits from the real data provider you want to use, and overrides the InsertEvent method adding your custom exception handling logic.

Check this: 
https://github.com/thepirat000/Audit.NET/issues/372

**Reply by shoregl** [reply]
This looks it will work for our requirements.  Thanks so much for the quick reply - and for sharing this excellent audit framework!
